datafusion/sql/src/planner.rs

```diff
@@ -189,8 +191,14 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 query,
                 with_options,
                 ..
-            } if columns.is_empty() && with_options.is_empty() => {
-                let plan = self.query_to_plan(*query, &mut HashMap::new())?;
+            } if with_options.is_empty() => {
+                let columns = if columns.is_empty() {
+                    None
+                } else {
+                    Some(columns)
+                };
+                let plan =
+                    self.query_to_plan(*query, &mut HashMap::new(), columns.as_ref())?;
```

Comment:
I'm not sure I understand why we don't have the projection logic here since this is specific to views, rather than passing the columns down into `query_to_plan`.

Reply:
> I'm not sure I understand why we don't have the projection logic here since this is specific to views, rather than passing the columns down into `query_to_plan`.

I could move the logic I added to the `query_to_plan_with_alias` method to be done after the `query_to_plan` call on line 201 if that's what you're thinking. My thought was simply that the plan should be fully formed on return from `query_to_plan`

Reply:
Yes, that's exactly what I'm suggesting. I think that `query_to_plan` is returning a valid plan for the query that is the input to the `CREATE VIEW` already. It seems appropriate that the code that is specific to creating the view should take care of adding the additional projection.

Reply:
Added this in the latest commit.